### PR TITLE
624 PACKAGE LEVEL pre-release activities

### PR DESCRIPTION
**Pull request by m7pr**
Part of #624 

* [x] Review and update README.md 
    * [x] check example code
    * [x] add CRAN BADGES
    * [x] update installation guidelines
    * [x] add Getting Help Section
* [x]  Review and update DESCRIPTION
    * [x] Package `Title` is not duplicated in Package `Description` in DESCRIPTION file (e.g. this happens in teal.slice currently)
    * [x] All package names in `Title` and `Description` fields of DESCRIPTION file are quoted with `'`
    * [x] LICENSE file is not mentioned in License field and LICENSE is add to `.Rbuildignore`
* [x] Run urlchecker::url_check() to identify broken links and fix
* [x] Run R CMD check --as-cran make sure everything pass
* [x] You have checked the Package Release Template https://github.com/insightsengineering/teal.reporter/pull/205/files
* [x] Make sure all `teal.*` mentions are lower-cased and quoted
* [x] Make Sure inst/WORDLIST is minimalized
* [x] Make sure each link to our documentation hosted with pkgdown on github pages do not have `/main/` in the address but it has `/latest/` instead, so we always expose the documentation of the latest release and not what's currently on main branch but not yet released
* [x] Remove exception in `.lintr: indentation_linter = NULL`
* [x] `@noMd` (in favor of `Roxygen: list(markdown = TRUE)` in `DESCRIPTION`)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.yml [hunk 1]
```diff
@@ -119,4 +119,4 @@ body:
       description: Any decision tree(s) that would aid release management
       placeholder: Any decision tree(s) that would aid release management.
       value: |
-        Click [here](https://github.com/insightsengineering/.github/blob/main/.github/ISSUE_TEMPLATE/RELEASE_DECISION_TREE.md) to see the release decision tree.
+        Click [here](https://github.com/insightsengineering/.github/blob/latest-tag/.github/ISSUE_TEMPLATE/RELEASE_DECISION_TREE.md) to see the release decision tree.
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply .github/ISSUE_TEMPLATE/release.yml:122]: they will be overwritten even if changed,  https://github.com/insightsengineering/teal.slice/pull/506#discussion_r1441497668

you could add this as an additional change here or can create separate issue https://github.com/insightsengineering/idr-tasks/issues/722

File: R/tm_data_table.R [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#' Data Table Viewer Teal Module
+#' Data Table Viewer `teal` Module
```
Comment (kartikeyakirar) [inline R/tm_data_table.R:1]: We maintain all titles and descriptions in sentence case, with titles not ending in a period.
```suggestion
#' Data table viewer `teal` module
```

File: R/tm_file_viewer.R [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#' File Viewer Teal Module
+#' File Viewer `teal` Module
```
Comment (kartikeyakirar) [inline R/tm_file_viewer.R:1]:
```suggestion
#' File viewer `teal` module
```

File: vignettes/using-association-plot.Rmd [hunk 4]
```diff
@@ -55,8 +55,8 @@ ADLB <- teal.modules.general::rADLB %>%
 
 ## Create an `app` variable
```
Comment (kartikeyakirar) [inline vignettes/using-association-plot.Rmd:56]:
```suggestion
## Initialize an app
```

File: .github/ISSUE_TEMPLATE/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (m7pr) [reply .github/ISSUE_TEMPLATE/release.yml:122]: I made a mistake in this change. This should not be here. In this context `latest-tag` is treated as branch :P not a tag/release.

File: R/tm_data_table.R (hunk 2 above)
Comment (m7pr) [reply R/tm_data_table.R:1]: @kartikeyakirar this PR does not cover this change.
I think other titles in all other files are title-case.
Would you mind creating a separate issue to change titles into sentenace case for tmg?

Also, would you like to take a lead on the same issue on tmc https://github.com/insightsengineering/teal.modules.clinical/issues/1073 ?

File: R/tm_file_viewer.R (hunk 3 above)
Comment (m7pr) [reply R/tm_file_viewer.R:1]: https://github.com/insightsengineering/teal.modules.general/pull/639#discussion_r1495682626

File: .github/ISSUE_TEMPLATE/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (m7pr) [reply .github/ISSUE_TEMPLATE/release.yml]: reverted f8d3583e828d0d92791db1b67462eff4ba1dfe98

File: R/tm_t_crosstable.R [hunk 5]
```diff
@@ -283,7 +283,7 @@ srv_t_crosstable <- function(id, data, reporter, filter_panel_api, label, x, y,
           substitute(
             expr = {
               lyt <- basic_tables %>%
-              split_call %>% # styler: off
+                split_call %>% # styler: off
```
[the remark this answers is not in the document]
Comment (m7pr) [reply R/tm_t_crosstable.R:337]: summoning @averissimo our lintr guru these days. Is this something that appears only here, or should we revisit the whole package? If yes, can we do it on a separate PR (pointing to this one)?

[the remark this answers is not in the document]
Comment (chlebowa) [reply R/tm_t_crosstable.R:337]: It's odd that it's even here, everything seems to follow the style guide.

[the remark this answers is not in the document]
Comment (pawelru) [reply R/tm_t_crosstable.R:337]: Spotted this because we touched that line. I think it was the reason why it got not styled. The change was good, the style had been applied and we probably don't need that line.
Haven't thought about the whole package. Checked this quickly and in my local repo (might be not in sync with pre-release branch) I found 7 occurences of `styler: off`. Some of them might be meaningful though.

[the remark this answers is not in the document]
Comment (m7pr) [reply R/tm_t_crosstable.R:337]: thanks @averissimo . makes sense now!

File: R/tm_file_viewer.R [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-#' @param input_path optional, (`list`) of the input paths to either: specific files of accepted formats,
-#'   a directory or a URL. The paths can be specified as absolute paths or relative to the running
-#'   directory of the application. Will default to current working directory if not supplied.
+#' @param input_path (`list`) of the input paths, optional. Each element can be:
 #'
-#' @export
+#' @inherit shared_params return
+#'
+#' Paths can be specified as absolute paths or relative to the running directory of the application.
+#' Default to the current working directory if not supplied.
```
Comment (averissimo) [inline R/tm_file_viewer.R:14]: This cannot be under `@inherit` tag!

```suggestion
#' Paths can be specified as absolute paths or relative to the running directory of the application.
#' Default to the current working directory if not supplied.
#'
#' @inherit shared_params return
```

Comment (m7pr) [reply R/tm_file_viewer.R:14]: ah, thanks

Comment (averissimo) [reply R/tm_file_viewer.R]: It was my bad, as I did the PR on this file... oopsie :laughing:

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 7]
```diff
@@ -33,23 +33,23 @@ body:
...
+      description: By submitting this issue, you agree to follow our [Contribution Guidelines.](https://insightsengineering.github.io/teal.modules.general/latest-tag/CONTRIBUTING.html)
       options:
         - label: I agree to follow this project's Contribution Guidelines.
           required: true
   - type: checkboxes
     id: security-policy
     attributes:
       label: Security Policy
-      description: By submitting this issue, you agree to follow our [Security Policy.](https://github.com/insightsengineering/teal.modules.general/security/policy)
+      description: By submitting this issue, you agree to follow our [Security Policy.](https://github.com/insightsengineering/teal.modules.general/latest-tag/security/policy)
```
Comment (kartikeyakirar) [inline .github/ISSUE_TEMPLATE/bug.yml:52]: I don't think we need to add latest-tag in templates . and this link does not exists

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 8]
```diff
@@ -13,23 +13,23 @@ body:
...
+      description: By submitting this issue, you agree to follow our [Contribution Guidelines.](https://insightsengineering.github.io/teal.modules.general/latest-tag/CONTRIBUTING.html)
       options:
         - label: I agree to follow this project's Contribution Guidelines.
           required: true
   - type: checkboxes
     id: security-policy
     attributes:
       label: Security Policy
-      description: By submitting this issue, you agree to follow our [Security Policy.](https://github.com/insightsengineering/teal.modules.general/security/policy)
+      description: By submitting this issue, you agree to follow our [Security Policy.](https://github.com/insightsengineering/teal.modules.general/latest-tag/security/policy)
```
Comment (kartikeyakirar) [inline .github/ISSUE_TEMPLATE/feature.yml:32]: https://github.com/insightsengineering/teal.modules.general/pull/639/files#r1507705411

File: .github/workflows/release.yaml [hunk 9]
```diff
@@ -54,7 +54,7 @@ jobs:
         checking Rd line widths .* NOTE
         checking top-level files .* NOTE
       unit-test-report-brand: >-
-        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
+        https://raw.githubusercontent.com/insightsengineering/hex-stickers/latest-tag/thumbs/teal.png
```
Comment (kartikeyakirar) [inline .github/workflows/release.yaml:57]: image

File: .pre-commit-config.yaml [hunk 10]
```diff
@@ -18,18 +18,18 @@ repos:
           - ggmosaic
           - ggplot2
           - shiny
-          - shinyTree
           - insightsengineering/teal
+          - insightsengineering/teal.transform
```
Comment (kartikeyakirar) [inline .pre-commit-config.yaml:22]:
```suggestion
          - teal
          - teal.transform
```
Shouldn't this be updated now that the packages are available on CRAN?

**Comment by m7pr** [comment]
Checks are failing due to a NOTE in R CMD CHECK. I see two NOTES 
image

**Comment by kartikeyakirar** [comment]
> Checks are failing due to a NOTE in R CMD CHECK. I see two NOTES

it was failing due to non-standard directory note. fixed it

**Comment by kartikeyakirar** [comment]
Hey @m7pr, some some minor comments above. Apart from that, the PR looks good to go after https://github.com/insightsengineering/teal.modules.general/pull/696.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 7 above)
Comment (m7pr) [reply .github/ISSUE_TEMPLATE/bug.yml:52]: Yes, sorry

File: .github/workflows/release.yaml (hunk 9 above)
Comment (m7pr) [reply .github/workflows/release.yaml]: e209bf3f4b20b8ade0e1cde2cf98c019b4bf9933

File: .github/ISSUE_TEMPLATE/feature.yml (hunk 8 above)
Comment (m7pr) [reply .github/ISSUE_TEMPLATE/feature.yml]: e209bf3f4b20b8ade0e1cde2cf98c019b4bf9933

**Comment by m7pr** [comment]
Thanks @kartikeyakirar for cleaning up the finish line! Everything is merged and lasts checks are running.
We just need to get an acceptance on this PR. Any last (famous) words :D?

File: .github/workflows/check.yaml [hunk 11]
```diff
@@ -40,7 +40,7 @@ jobs:
         checking for unstated dependencies in vignettes .* NOTE
         checking top-level files .* NOTE
       unit-test-report-brand: >-
-        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
+        https://raw.githubusercontent.com/insightsengineering/hex-stickers/thumbs/teal.png
```
Comment (kartikeyakirar) [inline .github/workflows/check.yaml:43]: revert this change as well. link is broken

Comment (averissimo) [reply .github/workflows/check.yaml:43]:
```suggestion
        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
```

Comment (m7pr) [reply .github/workflows/check.yaml:43]: thanks for the suggestion @averissimo : D I was about to make a suggestion on my own

File: .github/workflows/release.yaml [hunk 12]
```diff
@@ -54,7 +54,7 @@ jobs:
         checking Rd line widths .* NOTE
         checking top-level files .* NOTE
       unit-test-report-brand: >-
-        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
+        https://raw.githubusercontent.com/insightsengineering/latest-tag/thumbs/teal.png
```
Comment (averissimo) [inline .github/workflows/release.yaml:57]:
```suggestion
        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
```

Comment (m7pr) [inline .github/workflows/release.yaml:57]:
```suggestion
        https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
```

Comment (m7pr) [reply .github/workflows/release.yaml]: https://github.com/insightsengineering/teal.modules.general/pull/639/commits/8f95a49ee0bb9f2bab34ecfa676ec3ebdc0bd2bc

File: DESCRIPTION [hunk 13]
```diff
@@ -34,21 +36,22 @@ Imports:
     forcats (>= 1.0.0),
     grid,
     logger (>= 0.2.0),
-    magrittr (>= 1.5),
     scales,
     shinyjs,
+    shinyTree,
```
[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION:41]: And this comment https://github.com/insightsengineering/teal.modules.general/pull/663#discussion_r1499353649

[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION:41]: It should be at least `0.2.8` from what is stated in here https://github.com/insightsengineering/teal.modules.general/issues/181#issuecomment-934233440

File: vignettes/teal-modules-general.Rmd [hunk 14]
```diff
@@ -11,39 +10,41 @@ vignette: >
...
+The `teal.modules.general` package consist of collection of modules  essential for developing `teal` applications. 
+It is "general" in the sense that the intended functions of these modules are more fundamental. This contrasts with the more specialized focus on clinical data found in the `teal.modules.clinical` package. 
+The modules from `teal.modules.general`  can be used in conjunction with modules from `teal.modules.clinical` and / or
 other `shiny` modules to build a large `teal` / `shiny` app.
 
+
 The concepts presented here require knowledge about the core features of `teal`, specifically on how to launch a `teal`
 application and how to pass data into it. Therefore, it is highly recommended to refer to the [`README`](https://insightsengineering.github.io/teal/index.html) file and 
-the introductory [vignette](https://insightsengineering.github.io/teal/articles/teal.html) of the `teal` package.
+the introductory [vignette](https://insightsengineering.github.io/teal/latest-tag/articles/teal.html) of the `teal` package.
```
Comment (averissimo) [inline vignettes/teal-modules-general.Rmd:23]: This link is broken: https://insightsengineering.github.io/teal/latest-tag/articles/teal.html

**Review by averissimo — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 13 above)
Comment (m7pr) [inline DESCRIPTION:41]:
```suggestion
    shinyTree (>= 0.2.8),
```

[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION]: changed in here https://github.com/insightsengineering/teal.modules.general/pull/639/commits/f3981401f742e0afcba4063b4837d6cddf9d46a9

File: vignettes/teal-modules-general.Rmd (hunk 14 above)
Comment (m7pr) [inline vignettes/teal-modules-general.Rmd:23]:
```suggestion
the introductory [vignette](https://insightsengineering.github.io/teal/latest-tag/articles/getting-started-with-teal.html) of the `teal` package.
```

Comment (m7pr) [reply vignettes/teal-modules-general.Rmd]: 4dea7f3ae8a19c4f27d6d1f2c73d76fcc1ea0867

Comment (averissimo) [reply vignettes/teal-modules-general.Rmd]: Agree, I see you already done it :-)

**Review by averissimo — APPROVED** [review]
Looks good.

I went again through the initial bullet point list and it seems fine to merge!

File: .lintr [hunk 15]
```diff
@@ -2,5 +2,5 @@ linters: linters_with_defaults(
   line_length_linter = line_length_linter(120),
   cyclocomp_linter = NULL,
   object_usage_linter = NULL,
-  indentation_linter = NULL
+  object_name_linter = object_name_linter(styles = c("snake_case", "symbols"), regexes = c(ANL = "^ANL_?[0-9]*$", ADaM = "^r?AD[A-Z]{2,3}_?[0-9]*$"))
```
Comment (averissimo) [inline .lintr:5]: There are some `# nolint object_name` that can be removed from the code

Comment (averissimo) [reply .lintr:5]: I can whip a quick PR that removes them

image

Comment (chlebowa) [reply .lintr:5]: Maybe add `ADSL` and friends while you're at it?

Comment (averissimo) [reply .lintr:5]: Created here: https://github.com/insightsengineering/teal.modules.general/pull/703

Targets this branch

Comment (averissimo) [reply .lintr:5]: Number of `nolint` hints was drastically reduced

image

Comment (m7pr) [reply .lintr:5]: a true hero that we do not deserve

**Comment by m7pr** [comment]
Thanks @averissimo . I also did went again through the list, once you pointed some obvious mistakes!
Once https://github.com/insightsengineering/teal.modules.general/pull/703 is merged, I'll merge this one.

File: .lintr (hunk 15 above)
Comment (chlebowa) [reply .lintr]: Do you think you could do the same fot `tmc`?

**Comment by averissimo** [comment]
Fearless leader, we await the glorious merge :horse: :crossed_swords:

**Comment by m7pr** [comment]
veni, vidi, merge
